File: polocloud-suite/src/main/java/dev/httpmarco/polocloud/suite/commands/CommandSyntax.java
package dev.httpmarco.polocloud.suite.commands;

import dev.httpmarco.polocloud.suite.commands.type.KeywordArgument;

import java.util.Arrays;

public final class CommandSyntax {

    private final CommandExecution execution;
    private final CommandArgument<?>[] arguments;
    private  String description;

    public CommandSyntax(CommandExecution execution, CommandArgument<?>[] arguments, String description) {
        this.execution = execution;
        this.arguments = arguments;
    }

    public  String usage() {
        return String.join(" ", Arrays.stream(arguments)
                .map(it -> it instanceof KeywordArgument ? "&f" + it.key() : "&8<&f" + it.key() + "&8>")
                .toList()) +
                (description == null ? "" : " &8- &7" + description);
    }

    public CommandArgument<?>[] arguments() {
        return arguments;
    }

    public CommandExecution execution() {
        return execution;
    }
}

File: agent/src/main/java/dev/httpmarco/polocloud/agent/runtime/local/terminal/commands/CommandSyntax.kt
package dev.httpmarco.polocloud.agent.runtime.local.terminal.commands

import dev.httpmarco.polocloud.agent.runtime.local.terminal.commands.type.KeywordArgument
import java.util.Arrays

class CommandSyntax(
    val execution: CommandExecution,
    val description: String?,
    val arguments: MutableList<CommandArgument<*>>
) {
    fun usage(): String {
        return java.lang.String.join(
            " ", arguments.stream()
                .map { if (it is KeywordArgument) "&f" + it.key else "&8<&f" + it.key + "&8>" }
                .toList()
        ) + (if (description == null) "" else " &8- &7$description")
    }
}